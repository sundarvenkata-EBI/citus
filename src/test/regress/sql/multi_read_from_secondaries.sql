ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1600000;

SET citus.read_from_secondaries TO 'always';

CREATE TABLE the_table (a int, b int);

-- attempts to change metadata should fail while reading from secondaries
SELECT create_distributed_table('the_table', 'a');

SET citus.read_from_secondaries TO 'never';
SELECT create_distributed_table('the_table', 'a');

INSERT INTO the_table (a, b) VALUES (1, 1);
INSERT INTO the_table (a, b) VALUES (2, 1);

-- simluate actually having secondary nodes
SELECT * FROM pg_dist_node;
UPDATE pg_dist_node SET noderole = 'secondary';

SET citus.read_from_secondaries TO 'always';

-- inserts are disallowed
INSERT INTO the_table (a, b) VALUES (1, 2);

-- router selects are allowed
SELECT a FROM the_table WHERE a = 1;

-- real-time selects are not allowed
SELECT a FROM the_table;

SET citus.read_from_secondaries TO 'never';
UPDATE pg_dist_node SET noderole = 'primary';
DROP TABLE the_table;
