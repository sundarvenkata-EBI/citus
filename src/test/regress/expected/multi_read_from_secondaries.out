ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 1600000;
SET citus.read_from_secondaries TO 'always';
CREATE TABLE the_table (a int, b int);
-- attempts to change metadata should fail while reading from secondaries
SELECT create_distributed_table('the_table', 'a');
ERROR:  writing to worker nodes is not currently allowed
DETAIL:  citus.read_from_secondaries is set to 'always'
SET citus.read_from_secondaries TO 'never';
SELECT create_distributed_table('the_table', 'a');
 create_distributed_table 
--------------------------
 
(1 row)

INSERT INTO the_table (a, b) VALUES (1, 1);
INSERT INTO the_table (a, b) VALUES (2, 1);
SET citus.read_from_secondaries TO 'always';
-- inserts are disallowed
INSERT INTO the_table (a, b) VALUES (1, 2);
-- router selects are allowed
SELECT a FROM the_table WHERE a = 1;
 a 
---
 1
 1
(2 rows)

-- real-time selects are not allowed
SELECT a FROM the_table;
ERROR:  writing to worker nodes is not currently allowed
DETAIL:  citus.read_from_secondaries is set to 'always'
SET citus.read_from_secondaries TO 'never';
DROP TABLE the_table;
